enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java

```diff
@@ -165,925 +111,579 @@ public void shouldChangeUserPasswordIfSameUser() throws Exception
     }
 
-    //----------User suspension scenarios-----------
+    //---------- remove user from role -----------
 
     @Test
-    public void shouldSuspendUser() throws Exception
+    public void shouldRemoveUserFromRole() throws Exception
     {
-        testCallEmpty( db, adminSubject, "CALL dbms.suspendUser('readSubject')" );
-        assertTrue( manager.getUser( "readSubject" ).hasFlag( FileUserRealm.IS_SUSPENDED ) );
+        testCallEmpty( adminSubject, "CALL dbms.removeUserFromRole('readSubject', '" + READER + "')" );
+        assertFalse( "Should not have role reader", readSubject.getSubject().hasRole( READER ) );
     }
 
     @Test
-    public void shouldActivateUser() throws Exception
+    public void shouldKeepUserOutOfRole() throws Exception
     {
-        manager.suspendUser( "readSubject" );
-        testCallEmpty( db, adminSubject, "CALL dbms.activateUser('readSubject')" );
-        assertFalse( manager.getUser( "readSubject" ).hasFlag( FileUserRealm.IS_SUSPENDED ) );
+        assertFalse( "Should not have role publisher", readSubject.getSubject().hasRole( PUBLISHER ) );
+        testCallEmpty( adminSubject, "CALL dbms.removeUserFromRole('readSubject', '" + PUBLISHER + "')" );
+        assertFalse( "Should not have role publisher", readSubject.getSubject().hasRole( PUBLISHER ) );
     }
 
     @Test
-    public void shouldFailOnNonAdminSuspend() throws Exception
+    public void shouldFailToRemoveNonExistingUserFromRole() throws Exception
     {
-        testCallFail( db, schemaSubject, "CALL dbms.suspendUser('readSubject')",
-                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+        testCallFail( adminSubject, "CALL dbms.removeUserFromRole('Olivia', '" + PUBLISHER + "')",
+                QueryExecutionException.class, "User Olivia does not exist" );
+        testCallFail( adminSubject, "CALL dbms.removeUserFromRole('Olivia', 'thisRoleDoesNotExist')",
+                QueryExecutionException.class, "User Olivia does not exist" );
+        testCallFail( adminSubject, "CALL dbms.removeUserFromRole('Olivia', '')",
+                QueryExecutionException.class, "User Olivia does not exist" );
     }
 
     @Test
-    public void shouldFailOnNonAdminActivate() throws Exception
+    public void shouldFailToRemoveUserFromNonExistingRole() throws Exception
     {
-        manager.suspendUser( "readSubject" );
-        testCallFail( db, schemaSubject, "CALL dbms.activateUser('readSubject')",
-                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+        testCallFail( adminSubject, "CALL dbms.removeUserFromRole('readSubject', 'thisRoleDoesNotExist')",
+                QueryExecutionException.class, "Role thisRoleDoesNotExist does not exist" );
+        testCallFail( adminSubject, "CALL dbms.removeUserFromRole('readSubject', '')",
+                QueryExecutionException.class, "Role  does not exist" );
```

Comment:
Yes! There is a TODO mention of it (top of file), but I didn't fix it yet because I want to wait for a large piece of parallel rewriting on the production side.